data/data/openstack/main.tf

```diff
@@ -72,3 +72,21 @@ module "topology" {
   trunk_support       = var.openstack_trunk_support
   octavia_support     = var.openstack_octavia_support
 }
+
+resource "openstack_images_image_v2" "base_image" {
+  // we need to create a new image only if the base image url has been provided, plus base image name is <cluster_id>-rhcos
+  count = var.openstack_base_image_url != "" && var.openstack_base_image_name == "${var.cluster_id}-rhcos" ? 1 : 0
```

Comment:
this can be simplified to `var.openstack_base_image_url != ""` right?

Reply:
I'd rather leave it like that, just in case the user exports `TF_VAR_openstack_base_image_url`. We've had an issue in our CI where a previous version of this patch created a duplicate rhcos image, it caused all the jobs to fail.

Reply:
Yeah, this is an optional check to prevent possible regressions that can break our ci